logger/modules/log_file_merger_helpers.py

```diff
@@ -0,0 +1,228 @@
+        return False, None
+
+    # Read log entries from all log files
+    log_entries = []
+    for log_file in log_files:
+        try:
+            with log_file.open("r", encoding="utf-8") as file:
+                for line in file:
+                    try:
+                        # Attempt to parse the timestamp to ensure it matches the format
+                        datetime.strptime(line.split(": ")[0], log_datetime_format)
+                        log_entries.append(line)
+                    except ValueError:
+                        # Skip lines that do not match the format
+                        print(f"WARNING: Skipping invalid log entry: {line.strip()}")
+        except OSError:
+            # Skip files that cannot be opened
+            print(f"ERROR: Failed to read log file: {log_file}")
+
+    if len(log_entries) == 0:
+        print(f"ERROR: No log entries found in any log files in directory: {log_file_directory}")
+        return False, None
+
+    return True, log_entries
+
+
+def sort_log_entries(
+    log_entries: "list[str]", log_datetime_format: str
+) -> "tuple[bool, list[str] | None]":
+    """
+    Returns a sorted list of log entries in chronological order.
+
+    log_entries: An unsorted list of log entries.
+    log_datetime_format: Datetime format in the log entries.
+
+    Returns: Success, sorted list of log entries.
+    """
+    # Check that log entries is not empty
+    if len(log_entries) == 0:
+        print("ERROR: No log entries passed to sort function")
+        return False, None
+
+    # Sort the log entries based on the timestamp at the beginning of each entry
+    # The lambda function extracts the timestamp from each entry and converts it to a datetime object for comparison
+    sorted_log_entries = sorted(
+        log_entries, key=lambda entry: datetime.strptime(entry.split(": ")[0], log_datetime_format)
+    )
+
+    if len(sorted_log_entries) == 0:
+        print("ERROR: Failed to sort log entries")
+        return False, None
+
+    return True, sorted_log_entries
+
+
+def write_merged_logs(
+    sorted_log_entries: "list[str]", log_file_directory: pathlib.Path
+) -> "tuple[bool, pathlib.Path | None]":
+    """
+    Writes sorted logs to a logfile and returns the filepath.
+
+    sorted_log_entries: An sorted list of log entries.
+    log_file_directory: The directory where the merged log file will be written.
+
+    Returns: Success, path to merged log file.
+    """
+    # Check that sorted log entries is not empty
+    if len(sorted_log_entries) == 0:
+        print("ERROR: No log entries passed to write function")
+        return False, None
+
+    # Create the merged log file path
+    merged_log_file = pathlib.Path(log_file_directory, MERGED_LOGS_FILENAME)
+
+    # Write the log entries to the merged file
+    with merged_log_file.open("w", encoding="utf-8") as file:
+        file.writelines(sorted_log_entries)
```

Comment:
Is it possible for this to fail? Also, what if there already exists a `merged_logs.log` , does that just get overwritten?